docs: document user interactions (#610)
elastic/apm-agent-rum-js: docs/supported-technologies.asciidoc

[float]
[[user-interactions]]
=== User Interactions

The RUM agent automatically instruments click event listeners that are

> registered by your application. The click events are captured as `user-interaction`

Reviewer: ```suggestion
registered by the application and captured these events as `user-interaction`
```

> transactions. However, to avoid sending too many `user-interaction` transactions

Reviewer: Instead of saying it this way, We can explain like. Not all click events would be interesting and the agent would only capture the listeners as user-interactions when there is a network activity.

Author: I've rephrased it a bit.
